src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2795,11 +2795,9 @@ protected override void VisitStatement(BoundStatement statement)
 
         public override BoundNode? VisitUnconvertedObjectCreationExpression(BoundUnconvertedObjectCreationExpression node)
         {
-            var discardedDiagnostics = new DiagnosticBag();
-            var expr = _binder!.BindObjectCreationForErrorRecovery(node, discardedDiagnostics);
-            discardedDiagnostics.Free();
-            Visit(expr);
-            SetResultType(node, TypeWithState.Create(expr.Type, NullableFlowState.NotNull));
+            // This method is only involved in method inference with unbound lambdas.
+            // The diagnostics on arguments are reported by VisitObjectCreationExpression.
+            SetResultType(node, TypeWithState.Create(null, NullableFlowState.NotNull));
```

Comment:
>TypeWithState.Create(null, NullableFlowState.NotNull) [](start = 32, length = 53)

Does this create a `TypeWithState` with `Type == null` and `State == NotNull`? Should we pass `defaultState: MaybeNull` instead?

Reply:
Yes, that creates a `TypeWithState` with no type but `NoteNull` state.
Here we want to pass `NotNull` because we know it's not `null` (object creation always results in an instance) even if we don't know the type.

---
In reply to: [486468093](https://github.com/dotnet/roslyn/pull/47581#discussion_r486468093) [](ancestors = 486468093)